cmd/xl-storage.go

```diff
@@ -345,11 +351,39 @@ func (s *xlStorage) IsLocal() bool {
 	return true
 }
 
-func (s *xlStorage) Healing() bool {
+// Retrieve location indexes.
+func (s *xlStorage) GetDiskLoc() (poolIdx, setIdx, diskIdx int) {
+	// If unset, see if we can locate it.
+	if s.poolIndex < 0 || s.setIndex < 0 || s.diskIndex < 0 {
+		if api := newObjectLayerFn(); api != nil {
+			if ep, ok := api.(*erasureServerPools); ok {
+				if pool, set, disk, err := ep.getPoolAndSet(s.diskID); err == nil {
+					return pool, set, disk
+				}
+			}
+		}
+	}
```

Comment:
I am not really sure why have this kind of convoluted style? - can we not have a helper function that provides this at ObjectLayer() ? when a specific input is asked? such as 

```
diskID, err := disk.GetDiskID()
if err != nil {
}
ep.getPoolAndSet(diskID)
```

Instead?

Reply:
I feel it equally convoluted to add an interface function that just works for one instance.

Moved the code to a helper function.